packages/core/test/core/plan/change_validators/index.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+*                      Copyright 2021 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { AdapterOperations, ChangeValidator, ElemID, ObjectType, ReferenceExpression, toChange } from '@salto-io/adapter-api'
+import { mockFunction } from '@salto-io/test-utils'
+import { expressions } from '@salto-io/workspace'
+import getChangeValidators from '../../../../src/core/plan/change_validators'
+
+describe('getChangeValidators', () => {
+  it('should call both the adapter change validators and the core change validators', async () => {
+    const adapterChangeValidator = mockFunction<ChangeValidator>().mockResolvedValue([
+      {
+        elemID: new ElemID('adapter'),
+        message: 'message',
+        detailedMessage: '',
+        severity: 'Warning',
+      },
+    ])
+    const changesValidators = getChangeValidators({
+      adapter: {
+        deployModifiers: {
+          changeValidator: adapterChangeValidator,
+        },
+        fetch: jest.fn(),
+        deploy: jest.fn(),
+      },
+    } as Record<string, AdapterOperations>)
+
+    const type = new ObjectType({
+      elemID: new ElemID('adapter', 'type'),
+      annotations: {
+        value: new ReferenceExpression(new ElemID('adapter', 'someId'), new expressions.UnresolvedReference(new ElemID('adapter', 'someId'))),
+      },
+    })
+    const changes = [toChange({ after: type })]
+    const errors = await changesValidators.adapter(changes)
+    expect(errors.length).toBe(2)
```

Comment:
```suggestion
    expect(errors).toHaveLength(2)
```
this is important because when the test fails it would show the array and not just `1 != 2`